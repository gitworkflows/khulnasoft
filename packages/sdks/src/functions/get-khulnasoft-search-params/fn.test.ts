import { getKhulnasoftSearchParams } from './index.js';

const querystring =
  'someotherValue=jklsjfdal&abc=klfdjklgfds&khulnasoft.cachebust=true&khulnasoft.preview=page&khulnasoft.noCache=true&__khulnasoft_editing__=true&khulnasoft.overrides.page=037948e52eaf4743afed464f02c70da4&khulnasoft.overrides.037948e52eaf4743afed464f02c70da4=037948e52eaf4743afed464f02c70da4&khulnasoft.overrides.page%3A%2F=037948e52eaf4743afed464f02c70da4&preview_theme_id=128854393017';

const url = new URL(`localhost:3000/about-us?${querystring}`);

describe('Get Khulnasoft SearchParams', () => {
  test('correctly extracts all khulnasoft params from a query object', () => {
    const output = getKhulnasoftSearchParams(url.searchParams);
    expect(output).toMatchSnapshot();
  });
});
