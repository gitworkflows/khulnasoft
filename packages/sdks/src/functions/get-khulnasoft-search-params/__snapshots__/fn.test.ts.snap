// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Get Khulnasoft SearchParams > correctly extracts all khulnasoft params from a query object 1`] = `
{
  "cachebust": "true",
  "noCache": "true",
  "overrides.037948e52eaf4743afed464f02c70da4": "037948e52eaf4743afed464f02c70da4",
  "overrides.page": "037948e52eaf4743afed464f02c70da4",
  "overrides.page:/": "037948e52eaf4743afed464f02c70da4",
  "preview": "page",
}
`;
